TypeCobol/Compiler/Nodes/Attributes.cs

```diff
@@ -267,34 +302,55 @@ public object GetValue(object o, SymbolTable table)
             throw new System.ArgumentOutOfRangeException("Too many receiving items (" + variablesWritten.Count + ")");
         }
     }
-    internal class PointerReceiversAttribute : Attribute
+    internal class PointerDisplayableReceiversAttribute : Attribute
     {
         public object GetValue(object o, SymbolTable table)
         {
             var node = (Node)o;
             var codeElement = node.CodeElement;
             var setStatement = codeElement as SetStatementForIndexes;
-            List<List<DataDefinition>> variablesWrittenRaw = new List<List<DataDefinition>>();
-            List<DataDefinition> variablesWritten = new List<DataDefinition>();
+            List<string> displayableWritten = new List<string>();
             if (setStatement != null && node.IsFlagSet(Node.Flag.NodeContainsPointer))
             {
                 foreach (var data in setStatement.StorageAreaWrites)
                 {
-                    variablesWrittenRaw.Add(
-                        node.SymbolTable.DataEntries.Where(x => x.Key == data.MainSymbolReference.Name).SelectMany(x => x.Value).ToList());
+                    DataDescription dataDef = node.GetDataDefinitionForQualifiedName(data.MainSymbolReference.URI) as DataDescription;
+                    if (dataDef != null)
+                    {
+                        // Usage of Regex.Replace to replace only the first ooccurence of dataDef.Name to avoid probleme with groups like myPtrGroup::myPtr
+                        var regex = new Regex(Regex.Escape(dataDef.Name));
+                        displayableWritten.Add(
+                            regex.Replace(
+                                data.MainSymbolReference.ToString(), dataDef.Name + dataDef.Hash, 1)
+                                .Replace(" IN ", " OF "));
+                    }
                 }
-                
-                foreach (var varWritten in variablesWrittenRaw)
+
+            }
+            if (displayableWritten.Count == 0) return null;
+            return displayableWritten;
+            
+        }
+    }
+    internal class PointerReceiversAttribute : Attribute
+    {
+        public object GetValue(object o, SymbolTable table)
+        {
+            var node = (Node)o;
+            var codeElement = node.CodeElement;
+            var setStatement = codeElement as SetStatementForIndexes;
+            List<DataDefinition> variablesWritten = new List<DataDefinition>();
+            if (setStatement != null)
+            {
+                foreach (var data in setStatement.StorageAreaWrites)
```

Comment:
use the new StorageAreaWritesDataDefinition, you will get the DataDefinition directly.